src/wp-includes/class-wp-block-template.php

```diff
@@ -109,4 +110,12 @@ class WP_Block_Template {
 	 * @var bool
 	 */
 	public $is_custom = true;
+
+	/**
+	 * Author.
+	 *
+	 * @since 5.9.0
+	 * @var int
+	 */
+	public $post_author = 0;
```

Comment:
I think better to name this `$author`. That a template comes from a post is more of an implementation detail.